pkg/api/credentials.go

```diff
@@ -0,0 +1,32 @@
+package api
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+type LocalRef string
+
+// +k8s:deepcopy-gen=false
+
+// CredentialsProvider gives access to custom local credentials
+type CredentialsProvider interface {
+	Credentials() *corev1.LocalObjectReference
+}
+
+// +k8s:deepcopy-gen=false
+
+// ResourceWithCredentials is to be implemented by all CRDs using custom local credentials
+type ResourceWithCredentials interface {
+	CredentialsProvider
```

Comment:
Why do we need this spread across 2 interfaces, rather than just having the LocalObjectReference in this one?

Reply:
Because this can be implemented directly by the simple struct object, which is applied to the Spec sub-struct.

The wider interface applies to the actual CRD as a whole, not just the extended Spec.

It is the simplest way I found to make this as reusable as possible. The consumers just need to embed the local ref struct and implement the delegation to the interface it implements from the CRD struct.